### PR TITLE
refactor(datafusion): align register API across backends

### PR DESCRIPTION
**Pull request by gforsyth**
BREAKING CHANGE: `register_csv` and `register_parquet` are removed.
Pass filename to `register` method instead.

This also adds in a general backend test for the `register` API so we
can make sure things are consistent, and fixes a small bug with the
dispatch in polars (and datafusion which I copied) that raises a
recursion error if you pass in garbage.

### CODE REVIEW

**Comment by gforsyth** [comment]
I removed the argument and wording around passing `schema` for loading CSV files because datafusion doesn't accept `ibis.Schema` as an argument and I can't find any Python documentation online to point at in the docstring.

File: ibis/backends/tests/test_client.py [hunk 1]
```diff
@@ -845,3 +845,67 @@ def test_has_operation_no_geo(con):
...
+        "pyspark",
+        "snowflake",
+        "sqlite",
+    ]
+)
+def test_register_garbage(con):
+    with pytest.raises(ValueError) as exc_info:
+        con.register("garbage_notafile")
+
+    assert "Unrecognized file type or extension" in str(exc_info.value)
```
Comment (cpcloud) [inline ibis/backends/tests/test_client.py:911]:
```suggestion
    with pytest.raises(ValueError, match="Unrecognized file type or extension"):
        con.register("garbage_notafile")
```

**Review by cpcloud — APPROVED** [review]
LGTM, small and optional nit. Feel free to merge!

File: ibis/backends/tests/test_client.py (hunk 1 above)
Comment (gforsyth) [reply ibis/backends/tests/test_client.py:911]: oh, that's much nicer

**Pull request merged**